channeldb/channel.go

```diff
@@ -671,6 +688,81 @@ func (c *OpenChannel) MarkAsOpen(openLoc lnwire.ShortChannelID) error {
 	return nil
 }
 
+// MarkDataLoss marks sets the channel status to LocalDataLoss and stores the
+// passed commitPoint for use to retrieve funds in case the remote force closes
+// the channel.
+func (c *OpenChannel) MarkDataLoss(commitPoint *btcec.PublicKey) error {
+	c.Lock()
+	defer c.Unlock()
+
+	if err := c.Db.Update(func(tx *bolt.Tx) error {
+		chanBucket, err := updateChanBucket(
+			tx, c.IdentityPub, &c.FundingOutpoint, c.ChainHash,
+		)
+		if err != nil {
+			return err
+		}
+
+		channel, err := fetchOpenChannel(chanBucket, &c.FundingOutpoint)
+		if err != nil {
+			return err
+		}
+
+		channel.ChanStatus = LocalDataLoss
+
+		var b bytes.Buffer
+		if err := WriteElements(&b, commitPoint); err != nil {
+			return err
+		}
+
+		err = chanBucket.Put(dataLossCommitPointKey, b.Bytes())
+		if err != nil {
+			return err
+		}
+
+		return putOpenChannel(chanBucket, channel)
+	}); err != nil {
+		return err
+	}
+
+	// Update the in-memory representation to keep it in sync with the DB.
+	c.ChanStatus = LocalDataLoss
```

Comment:
do we ever check that we're not overwriting another subsystem's calls?

If we make `ChannelStatus` a bitvector, we can allow the calls to not overwrite each other

Reply:
Good call! I made it into a bitvector, and since the existing values used were 0, 1 and 2 (`0b10`), this worked out nicely with the new one being `4`.

Also made sure that all places we are reading this value wouldn't be affected, as we are just checking that it is not "Default".